nat-lab/tests/mesh_api.py

```diff
@@ -163,76 +195,90 @@ def get_meshmap(
-        self.assign_ip(
-            beta.id, "fd00::dead:2"
-        )  # TODO correct subnet when we'll decide about the range
-        return beta
-
-    def default_config_gamma_node(self, is_local: bool = False) -> Node:
-        gamma = self.register(
-            name="gamma",
-            node_id="39388b1e-ebd8-11ec-8ea0-0242ac120002",
-            private_key="GN+D2Iy9p3UmyBZhgxU4AhbLT6sxY0SUhXu0a0TuiV4=",
-            public_key="UnB+btGMEBXcR7EchMi28Hqk0Q142WokO6n313dt3mc=",
-            is_local=is_local,
-        )
-        self.assign_ip(gamma.id, "100.64.33.3")
-        self.assign_ip(
-            gamma.id, "fd00::dead:3"
-        )  # TODO correct subnet when we'll decide about the range
-        return gamma
-
     def default_config_two_nodes(
         self, alpha_is_local: bool = False, beta_is_local: bool = False
     ) -> Tuple[Node, Node]:
-        alpha = self.default_config_alpha_node(alpha_is_local)
-        beta = self.default_config_beta_node(beta_is_local)
-
-        alpha.set_peer_firewall_settings(beta.id, allow_incoming_connections=True)
-        beta.set_peer_firewall_settings(alpha.id, allow_incoming_connections=True)
-
-        return (alpha, beta)
+        alpha, beta, *_ = self.config_dynamic_nodes([alpha_is_local, beta_is_local])
+        alpha.set_peer_firewall_settings(beta.id, True)
+        beta.set_peer_firewall_settings(alpha.id, True)
+        return alpha, beta
 
     def default_config_three_nodes(
         self,
         alpha_is_local: bool = False,
         beta_is_local: bool = False,
         gamma_is_local: bool = False,
     ) -> Tuple[Node, Node, Node]:
-        (alpha, beta) = self.default_config_two_nodes(alpha_is_local, beta_is_local)
-        gamma = self.default_config_gamma_node(gamma_is_local)
-
+        alpha, beta, gamma, *_ = self.config_dynamic_nodes(
+            [alpha_is_local, beta_is_local, gamma_is_local]
+        )
         alpha.set_peer_firewall_settings(gamma.id, allow_incoming_connections=True)
+        alpha.set_peer_firewall_settings(beta.id, allow_incoming_connections=True)
+        beta.set_peer_firewall_settings(alpha.id, allow_incoming_connections=True)
         beta.set_peer_firewall_settings(gamma.id, allow_incoming_connections=True)
         gamma.set_peer_firewall_settings(alpha.id, allow_incoming_connections=True)
         gamma.set_peer_firewall_settings(beta.id, allow_incoming_connections=True)
-
-        return (alpha, beta, gamma)
+        return alpha, beta, gamma
 
     def _get_node(self, node_id: str) -> Node:
         if node_id not in self.nodes:
             raise MissingNodeError(node_id)
         return self.nodes[node_id]
+
+    @classmethod
+    def get_allowed_ip_list(cls, addrs: List[str]) -> List[str]:
+        return [
+            ip + "/32" if get_ip_address_type(ip) == IPProto.IPv4 else ip + "/128"
+            for ip in addrs
+            if get_ip_address_type(ip) in (IPProto.IPv4, IPProto.IPv6)
```

Comment:
Is it possible for `get_ip_address_type(...)` to return something that is not IPv4 nor IPv6?

Reply:
hm, actually no, exception will be thrown if anything else appears, so I guess this line is not needed

Reply:
ah sorry, it can return None, so it is needed actually

Reply:
improved this place abit